### PR TITLE
Update release tooling to trigger off a tag

### PR DESCRIPTION
**Pull request by grampelberg**

### CODE REVIEW

**Comment by grampelberg** [comment]
@Pothulapati as this is moving stuff around, there will need to be updates to the helm chart and docs, do you mind doing those as followups?

File: Makefile [hunk 1]
```diff
@@ -51,37 +55,43 @@ vendor:
...
+release: release-bootstrap build-chart
+ifndef CIRCLE_TAG
+	@echo "Missing CIRCLE_TAG, is this being run from circleci?"
+	@exit 1
+endif
+ifndef GITHUB_TOKEN
+	@echo "Requires a GITHUB_TOKEN with edit permissions for releases."
+	@exit 1
+endif
+	ghr -t ${GITHUB_TOKEN} \
```
Comment (Pothulapati) [inline Makefile]: We should hide this too right? as token would be visible otherwise

File: Makefile [hunk 2]
```diff
@@ -51,37 +55,43 @@ vendor:
...
-	set +e \
-    $(eval RELEASE_TAG=$(shell semantics --output-tag))
-	@if [ $(RELEASE_TAG) ]; then \
-	  echo "Tagging the latest commit image with tag: ${RELEASE_TAG}"; \
-	  docker tag ${IMAGE}:${COMMIT_TAG} ${IMAGE}:${RELEASE_TAG}; \
-	  echo "Pushing the docker image: ${IMAGE}:${RELEASE_TAG} "; \
-	  docker push ${IMAGE}:${RELEASE_TAG}; \
-	  echo "Generating binaries for Github Release"; \
-	  env GO111MODULE=on make vendor; \
-	  make binaries; \
```
Comment (michelleN) [inline Makefile]: how come you deleted the `make binaries` step?

Comment (grampelberg) [reply Makefile]: Felt a little weird to upload binaries only used inside the docker image that we already pushed.

File: Makefile (hunk 1 above)
Comment (grampelberg) [reply Makefile]: Security is officially hard. Now I need to go change that token!

File: Makefile (hunk 2 above)
Comment (michelleN) [reply Makefile]: ah yea 👍

**Comment by grampelberg** [comment]
Alright, take another look when y'all get the chance.

**Pull request merged**